app/client/websocket/index.ts

```diff
@@ -129,6 +133,11 @@ export default class WebSocketClient {
                 headers.Authorization = `Bearer ${this.token}`;
             }
             const {client} = await getOrCreateWebSocketClient(this.url, {headers, timeoutInterval: WEBSOCKET_TIMEOUT});
+
+            // Check again if the client is the same, to avoid race conditions
+            if (this.conn === client) {
```

Comment:
This should work, but I can't help think how this only works in a language like JS since it's single-threaded. In a multi-threaded language you'd need some sort of atomic action here to avoid a TOCTOU bug